### PR TITLE
[PUB-3728] Upgrade Storybook to latest version

### PR DESCRIPTION
**Pull request by guldenmw**
### Description

Upgrade Storybook to latest version. 

Includes:
- Replace our current Storybook configuration with latest recommended [main.js configuration](https://medium.com/storybookjs/declarative-storybook-configuration-49912f77b78).
- Migrate stories to new [Component Story Format (CSF)](https://storybook.js.org/blog/component-story-format/) format
- Migrate stories to use new [hierarchy separators](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#removed-hierarchy-separators)
- Update glob used to select stories as the previous one missed a few stories
- Remove `snapshot.test.js`
- Update necessary dependencies, `css-loader` and `postcss-loader` webpack config for `postcss` to work with Storybook

### Context & Notes

This PR is part of an attempt to keep up to date with dependency upgrades.
The Storybook snapshot tests currently aren't working as Jest might need updates. Will re-add `snapshot.test.js` as part of the Jest upgrade PR, but disabling snapshot tests for the time being.

### Best Practices Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] I have tested different user stories.
-   [ ] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

### ⚠️ Pre-deploy Checklist

- [x] I have manually loaded the URL of the staging deployment and confirmed it's working (see below**)
- [ ] I have confirmed that the Cypress E2E test is passing (_Note: we have a known issue where it doesn't run on some PRs. Hopefully this will be fixed soon._)
- [x] I have confirmed that all other tests are passing.

** To visit the URL of the staging deployment, please vlick the link in the bot comment, or look under checks at the bottom of this PR and click "Details" next to `bufferbotbrains/cicd-buffer-publish`
image

### CODE REVIEW

**Comment by guldenmw** [comment]
@anafilipadealmeida @MayaUribe @hamstu Would you mind just helping me confirm everything is still working fine as I've had to make minor adjustments to the css loaders in our webpack config and not sure where to really check the effects properly?

File: packages/date-time-preferences/components/DateTimePreferences/story.jsx [hunk 1]
```diff
@@ -9,21 +8,37 @@ import DateTimePreferences from './index';
...
-    />
-  ));
+export default {
+  title: 'DateTimePreferences',
+  decorators: [withA11y, (getStory) => <Provider store={store}>{getStory()}</Provider>],
+};
+
+export const AStory = () => <DateTimePreferences />;
+
+AStory.story = {
```
Comment (anafilipadealmeida) [inline packages/date-time-preferences/components/DateTimePreferences/story.jsx:18]: I know this name came from before, but it looks a bit weird 😂

**Review by anafilipadealmeida — APPROVED** [review]
I ran storybook and it seemed to be smooth, add-ons working, tests are passing, staging branch normal. Great job here @guldenmw! 👏  👏   And those simplified/new config files 😍

**Review by hamstu — DISMISSED** [review]
Awesome stuff @guldenmw! 

Looks like it's _almost_ perfect! There are two places where I know CSS is tricky in Publish. #1 is in the composer, but I clicked through there for a bit and didn't spy any obvious problems. The #2 place is the wonderful (* cough * not * cough *) Analyze components, aka the 'Analytics > Overview' tab. If you check there you can see the dropdown selects are broken:

image

My hunch is that these are broken since the .less files aren't being properly parsed anymore. It does seem that the `analyzeLessLoader` is still present in the webpack config, so that should be working right, but maybe it's the bump to a newer less version in the package.json that caused this? 🤔 

cc @anafilipadealmeida – here's analyze components continuing to bite us. 🙈

File: packages/date-time-preferences/components/DateTimePreferences/story.jsx (hunk 1 above)
Comment (guldenmw) [reply packages/date-time-preferences/components/DateTimePreferences/story.jsx:18]: 😂 so descriptive!

**Comment by anafilipadealmeida** [comment]
> My hunch is that these are broken since the .less files aren't being properly parsed anymore. It does seem that the `analyzeLessLoader` is still present in the webpack config, so that should be working right, but maybe it's the bump to a newer less version in the package.json that caused this? 🤔

Great catch here @hamstu! I had checked the composer and it seemed fine, but I totally ignored the analytics  🙈  We are so closeee!

**Comment by guldenmw** [comment]
@hamstu Good catch! Finally found the problem 🙂 `style-loader` updated version now sets the `esModule` option to true by default instead of false. Flipping it seems to do the trick! Also reverted `less-loader` version upgrade as it seems to only be relevant to analytics stuff anyway.

**Comment by anafilipadealmeida** [comment]
Had to be a bit creative to see analytics data @guldenmw , but it seems the css is working properly
Screenshot 2021-07-29 at 13 40 14
Screenshot 2021-07-29 at 13 22 40

**Pull request merged**